cmd/snap-recovery/partition/sfdisk.go

```diff
@@ -100,18 +100,14 @@ func (sf *SFDisk) Create(pv *gadget.LaidOutVolume) (map[string]string, error) {
 	}
 	buf, deviceMap := buildPartitionList(sf.partitionTable, pv)
 
-	// Write the partition table
+	// Write the partition table, note that sfdisk will re-read the
+	// partition table by itself: see disk-utils/sfdisk.c:write_changes()
 	cmd := exec.Command("sfdisk", sf.device)
 	cmd.Stdin = buf
 	if output, err := cmd.CombinedOutput(); err != nil {
 		return deviceMap, osutil.OutputErr(output, err)
 	}
 
-	// Reload the partition table using blockdev which is part of the initramfs
-	if output, err := exec.Command("blockdev", "--rereadpt", sf.device).CombinedOutput(); err != nil {
```

Comment:
Right, this only makes sense when we operate on the actual device and want to make the kernel reload the partition table. At some point perhaps osutil would even be a better place for this.

Reply:
Yeah, we don't need it here, I checked the sfdisk code and it is doing this by itself after it has written the changes to the disk.

Reply:
I also added a comment close to the sfdisk call that this already rescanns the partition table.